app.py

```diff
@@ -15,9 +15,150 @@
+)
+def cli(ctx=None, verbose=False) -> None:
+    """Thoth purge-job command line interface."""
+    if ctx:
+        ctx.auto_envvar_prefix = "THOTH_PURGE"
+
+    if verbose:
+        _LOGGER.setLevel(logging.DEBUG)
+
+    _LOGGER.debug("Debug mode is on")
+    _LOGGER.info("Component version: %s", __component_version__)
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    "--os-name",
+    "-o",
+    type=str,
+    metavar="OS_NAME",
+    envvar="THOTH_PURGE_OPERATING_SYSTEM_NAME",
+    required=True,
+    help="Purge specific operating system name data",
+)
+@click.option(
+    "--os-version",
+    "-v",
+    type=str,
+    metavar="OS_VERSION",
+    envvar="THOTH_PURGE_OPERATING_SYSTEM_VERSION",
+    required=True,
+    help="Purge specific operating system version data",
+)
+@click.option(
+    "--python-version",
+    "-p",
+    type=str,
+    metavar="PY_VERSION",
+    envvar="THOTH_PURGE_PYTHON_VERSION",
+    required=True,
+    help="Purge specific Python version data",
+)
+def solver(os_name: str, os_version: str, python_version: str) -> None:
+    """Purge solver data."""
+    graph = GraphDatabase()
+    graph.connect()
+    result = graph.delete_solved(os_name=os_name, os_version=os_version, python_version=python_version)
+    _LOGGER.info("Removed %d solver entries from the database", result)
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    "--adviser-version",
+    type=str,
+    metavar="VERSION",
+    envvar="THOTH_PURGE_ADVISER_VERSION",
+    help="Purge specific adviser results corresponding to a specific adviser version",
+)
+@click.option(
+    "--end-datetime",
+    "end",
+    type=str,
+    metavar="DATETIME",
+    envvar="THOTH_PURGE_ADVISER_END_DATETIME",
+    help="Purge specific adviser results older than datetime provided.",
+)
+def adviser(end: Optional[str], adviser_version: Optional[str]) -> None:
+    """Purge adviser data."""
+    graph = GraphDatabase()
+    graph.connect()
+    result = graph.delete_adviser_run(end_datetime=parse(end) if end else None, adviser_version=adviser_version)
+    _LOGGER.info("Removed %d adviser entries from the database", result)
+
+
+@cli.command()
+@click.pass_context
+@click.option(
+    "--package-extract-version",
+    type=str,
+    metavar="VERSION",
+    envvar="THOTH_PURGE_PACKAGE_EXTRACT_VERSION",
+    help="Purge specific package-extract results corresponding to a specific package-extract version",
+)
+@click.option(
+    "--end-datetime",
+    "end",
+    type=str,
+    metavar="DATETIME",
+    envvar="THOTH_PURGE_PACKAGE_EXTRACT_END_DATETIME",
+    help="Purge specific package-extract results older than datetime provided.",
+)
+def package_extract(end: Optional[str], package_extract_version: Optional[str]) -> None:
+    """Purge package-extract data."""
+    graph = GraphDatabase()
+    graph.connect()
+    result = graph.delete_package_extract_run(
+        end_datetime=parse(end) if end else None, package_extract_version=package_extract_version
```

Comment:
Hey Frido, I was looking at this code and storages. From what I can tell if no args are passed then all `package-extract`/`adviser-runs` will be purged from the database. Could we make `end-datetime` required? or is this behaviour okay in your opinion?

Reply:
Nice catch! I tried to enforce at least one filter on adapter level, but forgot to do it for package-extract: https://github.com/thoth-station/storages/pull/2302, PTAL

adviser-run should be fine:

https://github.com/thoth-station/storages/blob/24476513343296544db1bebadd67d8d0965018df/thoth/storages/graph/postgres.py#L6010-L6011